python/ray/dag/compiled_dag_node.py

```diff
@@ -704,7 +704,7 @@ def _preprocess(self) -> None:
             MultiOutputNode,
         )
 
-        self.input_task_idx, self.output_task_idx = None, None
```

Comment:
If we reset `output_task_idx` here, the assertion `assert self.output_task_idx is not None` will fail.